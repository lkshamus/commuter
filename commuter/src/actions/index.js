export const setCurrentWeather = (weather) => {
  return {
    type: 'SET_CURRENT_WEATHER',
    weather
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}

export const updateSearchWeather = (searchWeather) => {
  return {
    type: 'UPDATE_WEATHER_SEARCH',
    searchWeather
  }
}

export const setDirections = (directions) => {
  return {
    type: 'SET_DIRECTION',
    directions
  }
}

export const updateDirections = (searchedDirections) => {
  return {
    type: 'UPDATE_DIRECTIONS',
  searchedDirections
  }
}

export const setTransitDirections = (transitDirections) => {
  return {
    type: 'SET_TRANSIT_DIRECTIONS', 
    transitDirections
  }
}

export const setBikeDirections = (bikeDirections) => {
  return {
    type: 'SET_BIKE_DIRECTIONS',
    bikeDirections
  }
}

export const setWalkingDirections = (walkingDirections) => {
  return {
    type: 'SET_WALKING_DIRECTIONS', 
    walkingDirections
  }
}