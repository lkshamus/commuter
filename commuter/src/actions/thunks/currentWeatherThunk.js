import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { setCurrentWeather } from '../index'

export const getCurrentWeather = () => {
  return async (dispatch) => {
    // dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity);
      console.log(weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      // dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}