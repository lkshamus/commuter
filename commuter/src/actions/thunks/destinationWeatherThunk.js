import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, updateSearchWeather, setHasErrored, setHasFailed } from '../index'

export const getDestinationWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.hourlyWeatherByCity(city));
      if (weather === 'failed') {
        dispatch(setHasFailed(true));
      }
      dispatch(isLoading(false));
      return dispatch(updateSearchWeather(weather));
    } catch(error) {
      console.log(error.message)
      dispatch(setHasErrored(true));
    }
  }
}
