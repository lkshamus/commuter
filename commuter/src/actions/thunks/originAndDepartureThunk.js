// import fetchCall from '../../utilities/fetchCall';
// import * as cleaner from '../../utilities/cleaner';
// import { isLoading, setDirections } from '../index'

// export const getCurrentDrivingDirections = (origin, departure) => {
//   return async (dispatch) => {
//     dispatch(isLoading(true));
//     try {
//       const directions = await fetchCall(cleaner.orignAndDepartureDriving(origin, departure));
//       console.log('Driving', directions)
//       // if (weather === 'failed') {
//       //   dispatch(setIsOk(true));
//       // }
//       dispatch(isLoading(false));
//       return dispatch(setDirections(directions));
//     } catch(error) {
//       console.log('this is the error block')
//       console.log(error.message)
//       // dispatch(setHasErrored(true));
//     }
//   }
// }
