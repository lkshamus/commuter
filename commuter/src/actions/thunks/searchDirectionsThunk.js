import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, setDirections, setHasFailed, setHasErrored } from '../index'

export const getCurrentDirections = (origin, destination, mode) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
   try {
      const directions = await fetchCall(cleaner.orignAndDeparture(origin, destination, mode));
      if (directions === 'failed') {
        dispatch(setHasFailed(true));
      }
      dispatch(isLoading(false));
      return dispatch(setDirections(directions));
   } catch(error) {
      console.log('this is the error block')
      console.log(error.message)
     dispatch(setHasErrored(true));
   }
  }
}