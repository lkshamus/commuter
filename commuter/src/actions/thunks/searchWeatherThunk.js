import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, setCurrentWeather, updateSearchWeather } from '../index'

export const getCurrentWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity(city));
      console.log('weather results:', weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}
