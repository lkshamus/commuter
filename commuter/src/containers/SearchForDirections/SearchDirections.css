form {
  height: 400px;
  display: grid;
  grid-template-rows: 1fr 1fr;
  align-items: center;
  justify-content: center;
}

.origin, .destination {
  border-radius: 50px;
  font-size: 16px;
  grid-row: 1 / span 1;
  width: 400px;
  padding: 10px;
  margin: 25px;
  outline: none;
  border: .5px solid #c4d4e0;
}


ul {
  display: flex;
  width: 200px;
  text-align: center;
  justify-content: center;
  list-style: none;
}

li {
  padding: 50px;
}

.radio {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  position: relative;
  color: #666;
  top: 10px;
  height: 30px;
  width: 30px;
  border: 0;
  border-radius: 50px;
  cursor: pointer;     
  margin-right: 10px;
  outline: none;
}
.radio:checked::before {
  position: absolute;
  font: 13px/1 'Open Sans', sans-serif;
  left: 11px;
  top: 7px;
  content: '\02143';
  transform: rotate(40deg);
}
.radio:hover {
  background-color: #193446;

}

.radio:checked {
  background-color: #c4d4e0;
}

label {
  font: 300 16px/1.7 'Open Sans', sans-serif;
  color: #666;
  text-align: center;
  cursor: pointer;
  position: relative;
  right: 150px;
  bottom: 50px;
} 

.submit-btn {
  position: relative;
  background-color: #c4d4e0;
  bottom: 200px;
  left: 950px;
  color: #666;
  font: 300 16px/1.7 'Open Sans', sans-serif;
  border: 0;
  width: 90px;
  height: 43px;
  border-radius: 50px;
}

.submit-btn:hover {
  outline: none;
  color: #ffffff;
  background-color: #193446;
}

.submit-btn:disabled {
  background: #f1f1f1;
  color: #666;
}
