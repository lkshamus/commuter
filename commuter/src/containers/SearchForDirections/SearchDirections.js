import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { NavLink } from "react-router-dom";
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router'
import { Route } from 'react-router-dom';
import { Redirect } from 'react-router-dom'

// import { updateSearchWeather } from '../../actions'
import { getCurrentDirections } from '../../actions/thunks/searchDirectionsThunk'
// import { getCurrentDrivingDirections } from '../../actions/thunks/originAndDepartureThunk'
import { setDirections, updateDirections } from '../../actions'
import { getCurrentWeather } from '../../actions/thunks/searchWeatherThunk'
import { getDestinationWeather } from '../../actions/thunks/destinationWeatherThunk'
import { setCurrentWeather } from '../../actions'
import CurrentDirections from '../CurrentDirections/CurrentDirections'
import Loading from '../../components/Loading/index'
import './SearchDirections.css'

export class SearchDirections extends Component {
  constructor() {
    super()
    this.state = {
      origin: '',
      destination: '',
      mode: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  };

  handleRadioChange = (e) => {
  this.setState({
    mode: e.target.value
  });
}

  handleSubmitSearch = async (e) => {
    e.preventDefault();
    console.log('test')
    await this.props.getNewDirections(this.state.origin, this.state.destination, this.state.mode)
    let startCoordinates = this.props.directions.routes[0].legs[0].start_location 
    let endCoordinates = this.props.directions.routes[0].legs[0].end_location
    await this.props.displayWeatherStart(startCoordinates)
    await this.props.displayWeatherEnd(endCoordinates)
    this.props.history.push('/directions')
  };

  render() {
    return (
      <form
        onSubmit={this.handleSubmitSearch}>
        <input
        placeholder='enter origin' 
        className='origin'
        name='origin' 
        value={this.state.origin}
        onChange={(e) => this.handleChange(e)}
        />
        <input 
        placeholder='enter destination' 
        className='destination'
        name='destination' 
        value={this.state.destination}
        onChange={(e) => this.handleChange(e)}       
        />
       <button disabled={!this.state.mode}className='submit-btn'>submit</button>
        <ul>
          <li>
            <label>
              <input 
              className='radio'
              type='checkbox' 
              value='transit' 
              name='radio' 
              checked={this.state.mode === 'transit'}
              onChange={this.handleRadioChange}
              />
              transit
           </label>
          </li>
          <li>
            <label>
              <input 
              className='radio'
              type='checkbox' 
              value='walking' 
              name='radio'
              checked={this.state.mode === 'walking'}
              onChange={this.handleRadioChange}
            />
              walking
            </label>
          </li>
          <li>  
            <label>
              <input 
              className='radio'
              type='checkbox' 
              value='bicycling' 
              name='radio'
              checked={this.state.mode === 'bicycling'}
              onChange={this.handleRadioChange}
              />
              bicycling
            </label>
          </li> 
          <li> 
            <label>
              <input 
              className='radio'
              type='checkbox'  
              value='driving' 
              name='radio'
              checked={this.state.mode === 'driving'}
              onChange={this.handleRadioChange}
              />
              driving
            </label>
          </li>
        </ul>
      </form>
    )
  }
}


export const mapStateToProps = (state) => ({
  directions: state.directions,
  currentWeather: state.currentWeather,
  updateSearchWeather: state.destinationWeather,
  isLoading: state.isLoading
});

export const mapDispatchToProps = (dispatch) => ({
  getNewDirections: (origin, destination, mode) => dispatch(getCurrentDirections(origin, destination, mode)),
  displayWeatherStart: (city) => dispatch(getCurrentWeather(city)),
  displayWeatherEnd: (city) => dispatch(getDestinationWeather(city))
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchDirections);
