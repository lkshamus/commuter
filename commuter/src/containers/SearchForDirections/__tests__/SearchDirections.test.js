/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';

import { getCurrentDirections } from '../../../actions/thunks/searchDirectionsThunk'
import { SearchDirections, mapDispatchToProps, mapStateToProps } from '../SearchDirections';
import { setDirections } from '../../../actions'
import { setCurrentWeather } from '../../../actions'
import { updateSearchWeather } from '../../../actions'

describe('SearchDirections', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SearchDirections />)
    expect(wrapper).toMatchSnapshot();
  })
})

describe.skip('mapDispatchToProps', () => {
  it('calls dispatch with a setDirections action when getCurrentDirections is called', () => {

    const directions =  {origin: 'Mountain View', destination: 'San Jose'}
      
    const mockDispatch = jest.fn()
    const actionToDispatch = getNewDirections(directions)
      
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.displayDirections(directions)
      
    expect(mockDispatch).toHaveBeenCalled
  })

  it('calls dispatch with a setCurrentWeather action when getCurrentDirections is called', () => {

    const weather =  {city: 'Mountain View', temp: 70}
      
    const mockDispatch = jest.fn()
    const actionToDispatch = updateSearchWeather(weather)
      
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.getWeather(weather)
      
    expect(mockDispatch).toHaveBeenCalled
  })
})


describe('mapStateToProps', () => {
  it('should map the state to props', () => {

    const directions = {origin: 'Mountain View', destination: 'San Jose'}
    const startWeather = {city: 'Mountain View', temp: 70}

    const expected = {directions, startWeather}

    const mockState = {
      directions: directions,
      startWeather: startWeather
    }
    
    const mappedProps = mapStateToProps(mockState)
    expect (mappedProps).toEqual(expected)

  })
})
