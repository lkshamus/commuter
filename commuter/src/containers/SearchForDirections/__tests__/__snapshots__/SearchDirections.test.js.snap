// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDirections should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchDirections />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="origin"
          name="origin"
          onChange={[Function]}
          placeholder="enter origin"
          value=""
        />,
        <input
          className="departure"
          name="departure"
          onChange={[Function]}
          placeholder="enter departure"
          value=""
        />,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/directions"
        >
          <button
            className="submit-btn"
            disabled={true}
          >
            submit
          </button>
        </NavLink>,
        <ul>
          <li>
            <label>
              <input
                checked={false}
                className="radio"
                name="radio"
                onChange={[Function]}
                type="checkbox"
                value="transit"
              />
              transit
            </label>
          </li>
          <li>
            <label>
              <input
                checked={false}
                className="radio"
                name="radio"
                onChange={[Function]}
                type="checkbox"
                value="walking"
              />
              walking
            </label>
          </li>
          <li>
            <label>
              <input
                checked={false}
                className="radio"
                name="radio"
                onChange={[Function]}
                type="checkbox"
                value="bicycling"
              />
              bicycling
            </label>
          </li>
          <li>
            <label>
              <input
                checked={false}
                className="radio"
                name="radio"
                onChange={[Function]}
                type="checkbox"
                value="driving"
              />
              driving
            </label>
          </li>
        </ul>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "origin",
          "name": "origin",
          "onChange": [Function],
          "placeholder": "enter origin",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "departure",
          "name": "departure",
          "onChange": [Function],
          "placeholder": "enter departure",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="submit-btn"
            disabled={true}
          >
            submit
          </button>,
          "to": "/directions",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "submit",
            "className": "submit-btn",
            "disabled": true,
          },
          "ref": null,
          "rendered": "submit",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="transit"
                />
                transit
              </label>
            </li>,
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="walking"
                />
                walking
              </label>
            </li>,
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="bicycling"
                />
                bicycling
              </label>
            </li>,
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="driving"
                />
                driving
              </label>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="transit"
                />
                transit
              </label>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="transit"
                  />,
                  "transit",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "radio",
                    "name": "radio",
                    "onChange": [Function],
                    "type": "checkbox",
                    "value": "transit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "transit",
              ],
              "type": "label",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="walking"
                />
                walking
              </label>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="walking"
                  />,
                  "walking",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "radio",
                    "name": "radio",
                    "onChange": [Function],
                    "type": "checkbox",
                    "value": "walking",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "walking",
              ],
              "type": "label",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="bicycling"
                />
                bicycling
              </label>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="bicycling"
                  />,
                  "bicycling",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "radio",
                    "name": "radio",
                    "onChange": [Function],
                    "type": "checkbox",
                    "value": "bicycling",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "bicycling",
              ],
              "type": "label",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="driving"
                />
                driving
              </label>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="driving"
                  />,
                  "driving",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "radio",
                    "name": "radio",
                    "onChange": [Function],
                    "type": "checkbox",
                    "value": "driving",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "driving",
              ],
              "type": "label",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="origin"
            name="origin"
            onChange={[Function]}
            placeholder="enter origin"
            value=""
          />,
          <input
            className="departure"
            name="departure"
            onChange={[Function]}
            placeholder="enter departure"
            value=""
          />,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/directions"
          >
            <button
              className="submit-btn"
              disabled={true}
            >
              submit
            </button>
          </NavLink>,
          <ul>
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="transit"
                />
                transit
              </label>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="walking"
                />
                walking
              </label>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="bicycling"
                />
                bicycling
              </label>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="radio"
                  name="radio"
                  onChange={[Function]}
                  type="checkbox"
                  value="driving"
                />
                driving
              </label>
            </li>
          </ul>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "origin",
            "name": "origin",
            "onChange": [Function],
            "placeholder": "enter origin",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "departure",
            "name": "departure",
            "onChange": [Function],
            "placeholder": "enter departure",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="submit-btn"
              disabled={true}
            >
              submit
            </button>,
            "to": "/directions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "submit",
              "className": "submit-btn",
              "disabled": true,
            },
            "ref": null,
            "rendered": "submit",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="transit"
                  />
                  transit
                </label>
              </li>,
              <li>
                <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="walking"
                  />
                  walking
                </label>
              </li>,
              <li>
                <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="bicycling"
                  />
                  bicycling
                </label>
              </li>,
              <li>
                <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="driving"
                  />
                  driving
                </label>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="transit"
                  />
                  transit
                </label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      checked={false}
                      className="radio"
                      name="radio"
                      onChange={[Function]}
                      type="checkbox"
                      value="transit"
                    />,
                    "transit",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "checked": false,
                      "className": "radio",
                      "name": "radio",
                      "onChange": [Function],
                      "type": "checkbox",
                      "value": "transit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "transit",
                ],
                "type": "label",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="walking"
                  />
                  walking
                </label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      checked={false}
                      className="radio"
                      name="radio"
                      onChange={[Function]}
                      type="checkbox"
                      value="walking"
                    />,
                    "walking",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "checked": false,
                      "className": "radio",
                      "name": "radio",
                      "onChange": [Function],
                      "type": "checkbox",
                      "value": "walking",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "walking",
                ],
                "type": "label",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="bicycling"
                  />
                  bicycling
                </label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      checked={false}
                      className="radio"
                      name="radio"
                      onChange={[Function]}
                      type="checkbox"
                      value="bicycling"
                    />,
                    "bicycling",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "checked": false,
                      "className": "radio",
                      "name": "radio",
                      "onChange": [Function],
                      "type": "checkbox",
                      "value": "bicycling",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "bicycling",
                ],
                "type": "label",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <label>
                  <input
                    checked={false}
                    className="radio"
                    name="radio"
                    onChange={[Function]}
                    type="checkbox"
                    value="driving"
                  />
                  driving
                </label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      checked={false}
                      className="radio"
                      name="radio"
                      onChange={[Function]}
                      type="checkbox"
                      value="driving"
                    />,
                    "driving",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "checked": false,
                      "className": "radio",
                      "name": "radio",
                      "onChange": [Function],
                      "type": "checkbox",
                      "value": "driving",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "driving",
                ],
                "type": "label",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
